File: app/src/main/java/com/artemkaxboy/android/autoredialce/WidgetLast.java
package com.artemkaxboy.android.autoredialce;

import android.app.PendingIntent;
import android.appwidget.AppWidgetManager;
import android.appwidget.AppWidgetProvider;
import android.content.Context;
import android.content.Intent;
import android.widget.RemoteViews;

public class WidgetLast extends AppWidgetProvider {
//	public static final String WIDGET_IDS_KEY = "autoredialWidgetStatus";
	public static final String FROM_WIDGET = "from_widget";

	@Override
	public void onUpdate( Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds ) {
		super.onUpdate( context, appWidgetManager, appWidgetIds );
		RemoteViews remoteViews = new RemoteViews( context.getPackageName(), R.layout.widget_last );
		Intent intent = new Intent( ReceiverCommand.ACTION_REDIALING_START ).putExtra( FROM_WIDGET, true );
		PendingIntent pendingIntent = PendingIntent.getBroadcast( context, 42, intent, 0 );
		remoteViews.setOnClickPendingIntent( R.id.widgetImage, pendingIntent );
		appWidgetManager.updateAppWidget( appWidgetIds, remoteViews );
	}
}

File: app/src/main/java/com/artemkaxboy/android/autoredialce/WidgetLast.kt
package com.artemkaxboy.android.autoredialce

import android.app.PendingIntent
import android.appwidget.AppWidgetManager
import android.appwidget.AppWidgetProvider
import android.content.Context
import android.widget.RemoteViews

class WidgetLast : AppWidgetProvider() {

    override fun onUpdate(
        context: Context,
        appWidgetManager: AppWidgetManager,
        appWidgetIds: IntArray
    ) {
        val intent = ReceiverCommand.getIntent(context,
                ReceiverCommand.ACTION_REDIALING_START).putExtra(FROM_WIDGET, true)
        val pendingIntent = PendingIntent.getBroadcast(context, REQUEST_CODE, intent, 0)
        val remoteViews = RemoteViews(context.packageName, R.layout.widget_last)
        remoteViews.setOnClickPendingIntent(R.id.widgetImage, pendingIntent)
        appWidgetManager.updateAppWidget(appWidgetIds, remoteViews)
    }

    companion object {
        const val FROM_WIDGET = "from_widget"
        const val REQUEST_CODE = 42
    }
}
